packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -578,18 +579,42 @@ export const Table = <
     );
   }, [data, getSubRows]);
 
-  const [rowHasParent] = React.useState(() => new WeakSet<T>());
-
   const getSubRowsWithSubComponents = React.useCallback(
-    (originalRow: T) => {
-      if (!rowHasParent.has(originalRow)) {
-        rowHasParent.add(originalRow);
-        return [originalRow];
+    (originalRow: T, relativeIndex: number) => {
+      const symbol = Symbol.for('iui-id');
```

Comment:
I think we want to use a private a symbol rather than a global one. To create a private symbol, use the [`Symbol()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/Symbol) constructor, rather than [`Symbol.for()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/for), and save a reference to it.

So at the top of the file, you can declare this:

```js
const iuiId = Symbol('iui-id');
```

and use it everywhere in the file, e.g. `originalRow[iuiId]`

Reply:
I have made the change.

Reply:
It needs to be outside the component, otherwise a different symbol will be re-created on every render.

Reply:
I have moved it outside.